src/pages-and-resources/discussions/app-config-form/apps/legacy/LegacyConfigForm.jsx

```diff
@@ -15,48 +16,79 @@ import AppConfigFormDivider from '../shared/AppConfigFormDivider';
 function LegacyConfigForm({
   appConfig, onSubmit, formRef, intl, title,
 }) {
-  const {
-    handleSubmit,
-    handleChange,
-    handleBlur,
-    values,
-    errors,
-  } = useFormik({
-    initialValues: appConfig,
-    validationSchema: Yup.object().shape({
-      blackoutDates: Yup.string().matches(
-        blackoutDatesRegex,
-        intl.formatMessage(messages.blackoutDatesFormattingError),
-      ),
-    }),
-    onSubmit,
-  });
-
   return (
-    <Card className="mb-5 px-4 px-sm-5 pb-5" data-testid="legacyConfigForm">
-      <Form ref={formRef} onSubmit={handleSubmit}>
-        <h3 className="text-primary-500 my-3">{title}</h3>
-        <AppConfigFormDivider thick />
-        <AnonymousPostingFields
-          onBlur={handleBlur}
-          onChange={handleChange}
-          values={values}
-        />
-        <AppConfigFormDivider />
-        <DivisionByGroupFields
-          onBlur={handleBlur}
-          onChange={handleChange}
-          values={values}
-        />
-        <AppConfigFormDivider thick />
-        <BlackoutDatesField
-          errors={errors}
-          onBlur={handleBlur}
-          onChange={handleChange}
-          values={values}
-        />
-      </Form>
-    </Card>
+    <Formik
+      initialValues={appConfig}
+      validationSchema={
+        Yup.object().shape({
+          blackoutDates: Yup.string().matches(
+            blackoutDatesRegex,
+            intl.formatMessage(messages.blackoutDatesFormattingError),
+          ),
+          discussionTopics: Yup.array()
+            .of(
+              Yup.object().shape({
+                name: Yup.string().required(intl.formatMessage(messages.discussionTopicRequired)),
+              }),
+            ).test('unique', (
+              list,
+              testContext,
+              message = intl.formatMessage(messages.discussionTopicNameAlreadyExist),
+            ) => {
+              const mapper = x => x.name;
```

Comment:
A simpler way would be to write a method that tries to finds a duplicate element and then returns the name of the duplicate. If it returns nothing then no duplicate

Reply:
Nit: use proper variable names instead of "i" or "l"